src/propagation/gauss_method.jl

```diff
@@ -0,0 +1,512 @@
+@auto_hash_equals struct GaussSolution{T <: Real, U <: Number}
```

Comment:
What about having this file/functionality under `observations/` dir? Or perhaps `preprocess_obs/` or `orbit_determination/`? (I think either of the latter are better.)

Reply:
Maybe even under a `module`...

Reply:
Referring now to this `struct`, please include some doctrings

Reply:
I've added a new folder `src/orbit_determination` which contains `osculating.jl` and `gauss_method.jl`. Also, `GaussSolution` now has a docstring.